pkg/secret/secret_test.go

```diff
@@ -1 +1,502 @@
 package secret
+
+import (
+	"testing"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func TestSecretPull(t *testing.T) {
+	testCases := []struct {
+		secretName         string
+		secretNamespace    string
+		expectedError       bool
+		addToRuntimeObjects bool
+		expectedErrorText   string
```

Comment:
Please add client bool args. Refer to: https://github.com/openshift-kni/eco-goinfra/blob/5966cfaaf0378411c7307befc09a20b1e59e1df2/pkg/metallb/adresspool_test.go#L44
and Add 1 more test case when client is false(nil)

Reply:
You added client but it's unused in test case.

Reply:
No, you have to add test case when you pass nil client to the Pull function and catch relevant error.
Please refer
https://github.com/openshift-kni/eco-goinfra/blob/5966cfaaf0378411c7307befc09a20b1e59e1df2/pkg/metallb/adresspool_test.go#L88